src/hotspot/share/runtime/vframe.cpp

```diff
@@ -226,7 +226,7 @@ void javaVFrame::print_lock_info_on(outputStream* st, int frame_count) {
       Klass* k = obj->klass();
       st->print_cr("\t- %s <" INTPTR_FORMAT "> (a %s)", "parking to wait for ", p2i(obj), k->external_name());
     }
-    else if (thread()->osthread()->get_state() == OBJECT_WAIT) {
+    else if (thread()->osthread()->get_state() == CONDVAR_WAIT) {
       // We are waiting on an Object monitor but Object.wait() isn't the
       // top-frame, so we should be waiting on a Class initialization monitor.
```

Comment:
Comment needs updating. I'm glad to see this still works okay with such a simple change.

Reply:
Oh, yes, it does need updating.